cmd/opentelemetry-prometheus-sidecar/main.go

```diff
@@ -157,20 +157,13 @@ func Main() bool {
 		return false
 	}
 
-	intervals, err := parseIntervals(cfg.Prometheus.ScrapeIntervals)
-	if err != nil {
-		level.Error(logger).Log("msg", "error parsing --prometheus.scrape-interval", "err", err)
-		return false
-	}
-
 	// Parse was validated already, ignore error.
 	promURL, _ := url.Parse(cfg.Prometheus.Endpoint)
 
-	readyCfg := config.PromReady{
-		Logger:          log.With(logger, "component", "prom_ready"),
-		PromURL:         promURL,
-		ScrapeIntervals: intervals,
-	}
+	promMon := prometheus.NewMonitor(config.PromReady{
```

Comment:
Note: There's some forward-looking code motion happening here. The Prometheus monitor object is becoming shared state that can be used across calls to inspect the WAL state. This means we can detect Prom crashes and handle them gracefully in the future by remembering the start time of the process being followed and re-initializing the scrape intervals and config settings after a Prom restart.